### ISSUE TITLE
[shared_preferences][Add-To-App][Multiple Entry Points]NoSuchMethodError: No top-level getter, when run on release mode using a different entry point 

### ISSUE THREAD

**Issue by AlvinaC** [flutter/flutter#103292]
## Steps to Reproduce

1. Create a flutter module and create any widget, for example here, 'Bookstore()' .
2. In the pubspec.yaml, give the latest version of shared preference as a dependancy.
`shared_preferences: ^2.0.13`
3. Give a different entry point other than main.
```dart
@pragma('vm:entry-point')
void toModuleA() async{
  WidgetsFlutterBinding.ensureInitialized();
  runApp(const Bookstore());
}
```
5. Run the module for sanity check and no errors. 
6. Build .aar for Add-To-App
`flutter build aar`
7. In a sample native android application, write the following in app level build.gradle
```dart
String storageUrl = System.env.FLUTTER_STORAGE_BASE_URL ?: "https://storage.googleapis.com"
repositories {
    maven {
        url '<path of the generated maven repository of .aar after step 6>'
    }
    maven {
        url "$storageUrl/download.flutter.io"
    }
}
```

9. In the dependancy mention the flutter .aar as the dependancy.
`releaseImplementation ('<.aar package path>:flutter_release:1.0')`
*.aar package path - will be displayed in the terminal of flutter module after step 6.
10. Create a application class for the native android application if not present and mention it's name in the manifest.xml.
example:
` <application
        android:name=".<your application name>"
   </application>`
11. Create an activity for launching flutter module.
```dart
class AFeatureActivity : io.flutter.embedding.android.FlutterActivity() {

    companion object {

        fun withCachedEngine(cachedEngineId: String): CachedEngineIntentBuilder {
            return CachedEngineIntentBuilder(AFeatureActivity::class.java, cachedEngineId)
        }

        fun withNewEngine(): NewEngineIntentBuilder {
            return NewEngineIntentBuilder(AFeatureActivity::class.java)
        }
    }

    var methodChannelInvoker: (MethodCall, MethodChannel.Result) -> Unit = { call, result ->

    }

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        Log.d("in onCreate", "in onCreate")
    }

    override fun onPostResume() {
        super.onPostResume()
        Log.d("in post resume", "in post resume")
    }

    override fun onResume() {
        super.onResume()
        Log.d("in  resume", "in  resume")
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        Log.d("in  onAttachedToWindow", "in  onAttachedToWindow")
    }

    override fun onFlutterSurfaceViewCreated(flutterSurfaceView: FlutterSurfaceView) {
        super.onFlutterSurfaceViewCreated(flutterSurfaceView)
        Log.d("onFlutterSurfaceView", "in  onFlutterSurfaceViewCreated")
    }

    override fun onFlutterUiDisplayed() {
        super.onFlutterUiDisplayed()
        Log.d("onFlutterUiDisplayed", "in  onFlutterUiDisplayed")
    }

    override fun configureFlutterEngine(flutterEngine: FlutterEngine) {
        super.configureFlutterEngine(flutterEngine)
        Log.d("configureFlutterEngine", "in  configureFlutterEngine")
        setUpMethodChannel()
    }

    private fun setUpMethodChannel() {
        val flutterEngine =
            (application as <your application name>).getCachedFlutterEngine()
        MethodChannel(
            flutterEngine.dartExecutor.binaryMessenger,
            <your application name>.CHANNEL
        ).setMethodCallHandler { call, result ->
            methodChannelInvoker(call, result)
        }
    }
}
```
13. Mention this activity in the manifest.xml. example:
```dart
 <activity
            android:name=".AFeatureActivity"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize" />
```
15. In the application class, write the following. Using a flutter engine group here for different entries to flutter module.
```dart
open class <your application name> : Application() {

    //flutter engine group
    lateinit var engines: FlutterEngineGroup

    //use pre cached flutter engine to minimize launch delay
    private lateinit var flutterEngineA: FlutterEngine
    
    override fun onCreate() {
        super.onCreate()
        
        //initFlutterEngine()
        initFlutterEngineGroup()
    }

    private fun initFlutterEngineGroup() {
        //engine grp init
        engines = FlutterEngineGroup(this)
        //entry point for feature A
        val dartEntrypointA =
            DartExecutor.DartEntrypoint(
                FlutterInjector.instance().flutterLoader().findAppBundlePath(), "toModuleA"
            )
        flutterEngineA = engines.createAndRunEngine(this, dartEntrypointA)
       
        FlutterEngineCache.getInstance().put(CACHED_FEATURE_A_FLUTTER_ENGINE_ID, flutterEngineA)
       
    }

    fun getCachedFlutterEngine(): FlutterEngine {
        return flutterEngineA
    }

    companion object {

        val CHANNEL = "<some channel name>/extras"
       
        val CACHED_FEATURE_A_FLUTTER_ENGINE_ID = "Flutter Feature A Engine"
       
    }

}
```
17. From anywhere in the application launch the activity created. example:
```dart
val intent = context?.let { it1 ->
                AFeatureActivity.withCachedEngine(<your application name>.CACHED_FEATURE_A_FLUTTER_ENGINE_ID)
                    .build(it1)
            }
            intent?.let { startActivity(intent) }
```
19. Run the android application in release mode.

**Expected results:** 
The flutter screen is seen.

**Actual results:** 
Blank screen is seen

<details>
<summary>Error in the console</summary>

When using the debugImplementation of the aar after including shared preference package, the flutter screen is seen. In the releaseImplementation, the blank screen shows up with the following error.

```console
 E/flutter: [ERROR:flutter/shell/common/shell.cc(94)] Dart Unhandled Exception: NoSuchMethodError: No top-level getter 'toModuleA' declared.
    Receiver: top-level
    Tried calling: toModuleA, stack trace: #0      NoSuchMethodError._throwNew (dart:core-patch/errors_patch.dart:218:5)
E/flutter: [ERROR:flutter/runtime/dart_isolate.cc(707)] Could not resolve main entrypoint function.
E/flutter: [ERROR:flutter/runtime/dart_isolate.cc(191)] Could not run the run main Dart entrypoint.
E/flutter: [ERROR:flutter/runtime/runtime_controller.cc(382)] Could not create root isolate.
E/flutter: [ERROR:flutter/shell/common/shell.cc(574)] Could not launch engine with configuration.`

```
</details>
<details>
<summary>flutter doctor -v</summary>

```console
`[✓] Flutter (Channel master, 2.6.0-12.0.pre.218, on Mac OS X 10.14.4 18E226 darwin-x64, locale en-GB)
    • Flutter version 2.6.0-12.0.pre.218 at /Users/macbookpro/FlutterDevelopment/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 39f5769cd8 (7 months ago), 2021-10-06 08:10:37 +0100
    • Engine revision 88e150ddc9
    • Dart version 2.15.0 (build 2.15.0-178.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/macbookpro/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[!] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ! CocoaPods 1.6.1 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] Connected device (2 available)
    • Pixel 2 XL (mobile) • 192.168.1.103:5555 • android-arm64  • Android 10 (API 29)
    • Chrome (web)        • chrome             • web-javascript • Google Chrome 101.0.4951.54

! Doctor found issues in 2 categories.
```
</details>

**Comment by stuartmorgan** [comment]
Duplicate of https://github.com/flutter/flutter/issues/91841

**Comment by AlvinaC** [comment]
[https://github.com/flutter/flutter/issues/91841#issuecomment-1027286001](url) Thanks this worked.
